packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -434,11 +434,6 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
       return false;
     }
 
-    if (resultCode != Activity.RESULT_OK) {
```

Comment:
https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInApi.html#getSignInResultFromIntent(android.content.Intent) says it can be null if you don't pass in the intent. The question then is whether the intent can be null or not. If the activity code is cancelled, it could mean the invoked intent crashed.

Translation: It is far from clear what the exact contract is and it looks flaky enough that we should guard against it. It is better to return PlatformException rather than a native crash IMO.